app/models/commit.rb

```diff
@@ -83,6 +83,19 @@ def refresh_statuses!
     end
   end
 
+  def add_status(status_attributes)
+    previous_status = significant_status
+    statuses.create!(status_attributes)
+    reload
```

Comment:
The `reload` is needed because the implementation of `signifcant_status` uses `||=` so `@significant_status` doesn't get refreshed after I've added it. I could just set `@significant_status = nil` instead of the `reload`.